qipr_approver/approver/tests/tests_models/test_approve_models.py

```diff
@@ -1,38 +1,108 @@
+        section1 = Section(name = 'section1', sort_order = 1)
+        section2 = Section(name = 'section2', sort_order = 1)
+
+        section1.save(self.user)
+        self.assertRaises(IntegrityError, lambda : section2.save(self.user))
+
+    def test_should_print_name_when_asked_for_string(self):
+        text = "sample text"
+        section = Section(name = text)
+        self.assertEqual(str(section), text)
 
 class QuestionTestCase(TestCase):
     def setUp(self):
-        pass
+        self.user = User(username = 'testuser')
+        self.user.save()
+
+    def test_should_pass_when_model_has_correct_fields(self):
+        self.assertTrue(is_foreign_key_to(Question, "section", Section))
+        self.assertTrue(check_fields(Question, "text", "Text"))
+        self.assertTrue(check_fields(Question, "description", "Text"))
+        self.assertTrue(check_fields(Question, "sort_order", "Integer"))
+        self.assertTrue(is_foreign_key_to(Question, "correct_choice", Choice))
+        self.assertTrue(is_foreign_key_to(Question, "choice", Choice))
+
+    def test_should_throw_exception_when_saved_without_correct_choice(self):
+        question = Question()
+        self.assertRaises(IntegrityError, lambda : question.save(self.user))
 
-    def test_no_question_tests_written(self):
-        self.fail('You have not written any tests for {}'.format(type(self)))
+    def test_should_pass_when_saving_with_user_and_correct_choice(self):
+        question = Question()
+        choice = Choice(text='testChoice', sort_order = 1)
+        choice.save(self.user)
+        question.correct_choice = choice
+        question.save(self.user)
+        self.assertTrue(question in Question.objects.all())
+
+    def test_should_print_text_when_asked_for_string(self):
+        text = "sample choice"
+        question = Question(text = text)
+        self.assertEqual(str(question), text)
 
 class ChoiceTestCase(TestCase):
     def setUp(self):
-        pass
+        self.user = User(username = 'testuser')
+        self.user.save()
+
+    def test_should_pass_when_model_has_correct_fields(self):
+        self.assertTrue(check_fields(Choice, "text", "Text"))
+        self.assertTrue(check_fields(Choice, "sort_order", "Integer"))
+
+    def test_should_throw_exception_when_not_unique_sort_order(self):
+        choice1 = Choice(text = 'choice1', sort_order = 1)
+        choice2 = Choice(text = 'choice2', sort_order = 1)
 
-    def test_no_choice_tests_written(self):
-        self.fail('You have not written any tests for {}'.format(type(self)))
+        choice1.save(self.user)
+        self.assertRaises(IntegrityError, lambda : choice2.save(self.user))
+
+    def test_should_print_text_when_asked_for_string(self):
+        text = "sample choice"
+        choice = Choice(text = text)
+        self.assertEqual(str(choice), text)
 
 class ResponseTestCase(TestCase):
     def setUp(self):
-        pass
+        self.user = User(username = 'testuser')
```

Comment:
I'm seeing these two lines a lot. If you have common setup operations you can inherit from another class that does that. 

I.E. instead of ResponseTestCase(TestCase) do something like ResponseTestCase(SetupUserTestCase)